src/plugins/chartjs/timeline/timeline-chart.module.ts

```diff
@@ -385,6 +387,138 @@ export class TimelineChartPlugin {
         }
     }
 
+    private handleMouseMove(chart: TimelineChart, event: Partial<MouseEvent>): void {
+        const mousePosition = this.isWithinHandle(chart, event);
+
+        let timelineOptions = chart.options as TimelineChartOptions;
+        let hasTooltipOnRange: boolean = timelineOptions.timeline.range.hasOwnProperty('tooltip');
+        let hasTooltipOnHandles: boolean = timelineOptions.timeline.handles.hasOwnProperty('tooltip');
+        let timelineTooltipText: string;
+        let handleTooltipText: { rangeLower: string, rangeUpper: string };
+
+        if (hasTooltipOnRange) {
+            timelineTooltipText = timelineOptions.timeline.range.tooltip.label();
+        }
+
+        if (hasTooltipOnHandles) {
+            handleTooltipText = timelineOptions.timeline.handles.tooltip.label();
+        }
+
+        if (mousePosition === TimelineHandle.Range && hasTooltipOnRange) {
+            this.externalTooltipHandler(chart, TimelineHandle.Range, timelineTooltipText);
+        } else if (mousePosition === TimelineHandle.Lower && hasTooltipOnHandles) {
+            this.externalTooltipHandler(chart, TimelineHandle.Lower, handleTooltipText.rangeLower);
+        } else if (mousePosition === TimelineHandle.Upper && hasTooltipOnHandles) {
+            this.externalTooltipHandler(chart, TimelineHandle.Upper, handleTooltipText.rangeUpper);
+        } else {
+            let tooltipEl = this.getOrCreateTooltip(chart);
+            tooltipEl.style.opacity = '0';
+        }
+
+    }
+
+
+    private getOrCreateTooltip(chart: TimelineChart) {
+        let tooltipEl = chart.canvas.parentNode.querySelector('.timeline-tooltip') as HTMLElement;
+
+        if (!tooltipEl) {
+            tooltipEl = document.createElement('div');
+            tooltipEl.classList.add('timeline-tooltip');
+            tooltipEl.classList.add('tooltip');
+
+            tooltipEl.style.zIndex = '1';
```

Comment:
Do these styles need to be hard-coded, or can they be part of the `.timeline-tooltip` CSS rule?

Reply:
I added them into the `timeline-chart.component.less` but had to use `::ng-deep` to get them to be picked up

Reply:
You should be able to do it without ng-deep by adding it to tooltips.less.